projects/addon-doc/src/components/language-switcher/language-switcher.component.ts

```diff
@@ -0,0 +1,32 @@
+import {ChangeDetectionStrategy, Component, Inject} from '@angular/core';
+import {FormControl} from '@angular/forms';
+import {TuiLanguageName, TuiLanguageSwitcher} from '@taiga-ui/i18n';
+
+@Component({
+    selector: `tui-language-switcher`,
+    templateUrl: `./language-switcher.component.html`,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiLanguageSwitcherComponent {
+    language = new FormControl(this.switcher.currentLanguage);
+
+    flags: Map<TuiLanguageName, string> = new Map([
+        [`Chinese`, `🇨🇳 chinese`],
+        [`Dutch`, `🇳🇱 dutch`],
+        [`English`, `🇬🇧 english`],
+        [`French`, `🇫🇷 french`],
+        [`German`, `🇩🇪 german`],
+        [`Italian`, `🇩🇪 italian`],
+        [`Polish`, `🇵🇱 polish`],
+        [`Portuguese`, `🇵🇹 portuguese`],
+        [`Russian`, `🇷🇺 russian`],
+        [`Spanish`, `🇪🇸 spanish`],
+        [`Turkish`, `🇹🇷 turkish`],
+        [`Ukrainian`, `🇺🇦 ukrainian`],
+        [`Vietnamese`, `🇻🇳 vietnamese`],
+    ]);
+
+    names: TuiLanguageName[] = Array.from(this.flags.keys());
```

Comment:
nit: all props in this component are `readonly` ;)

Reply:
fixed